core/module_specifier.rs

```diff
@@ -80,3 +134,256 @@ impl PartialEq<String> for ModuleSpecifier {
+      ModuleSpecifier::resolve_from_cwd(
+        "https://deno.land/core/tests/006_url_imports.ts"
+      ).unwrap()
+      .to_string(),
+      "https://deno.land/core/tests/006_url_imports.ts",
+    );
+  }
+
+  #[test]
+  fn test_resolve() {
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "./005_more_imports.ts",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "http://deno.land/core/tests/005_more_imports.ts"
+    );
+
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "../005_more_imports.ts",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "http://deno.land/core/005_more_imports.ts"
+    );
+
+    // absolute URL
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "http://deno.land/core/tests/005_more_imports.ts",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "http://deno.land/core/tests/005_more_imports.ts",
+    );
+
+    // absolute path
+    if cfg!(target_os = "windows") {
+      assert_eq!(
+        ModuleSpecifier::resolve(
+          r"C:/deno/tests/005_more_imports.ts",
+          r"C:/deno/tests/006_url_imports.ts",
+        ).unwrap()
+        .to_string(),
+        "file:///C:/deno/tests/005_more_imports.ts",
+      );
+      assert_eq!(
+        ModuleSpecifier::resolve(
+          r"C:\deno\tests\005_more_imports.ts",
+          r"C:/deno/tests/006_url_imports.ts",
+        ).unwrap()
+        .to_string(),
+        "file:///C:/deno/tests/005_more_imports.ts",
+      );
+    } else {
+      assert_eq!(
+        ModuleSpecifier::resolve(
+          "/dev/core/tests/005_more_imports.ts",
+          "/dev/core/tests/006_url_imports.ts",
+        ).unwrap()
+        .to_string(),
+        "file:///dev/core/tests/005_more_imports.ts",
+      );
+    }
+
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "data:text/javascript,export default 'grapes';",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "data:text/javascript,export default 'grapes';",
+    );
+
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "blob:https://whatwg.org/d0360e2f-caee-469f-9a2f-87d5b0456f6f",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "blob:https://whatwg.org/d0360e2f-caee-469f-9a2f-87d5b0456f6f",
+    );
+
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "javascript:export default 'artichokes';",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "javascript:export default 'artichokes';",
+    );
+
+    assert_eq!(
+      ModuleSpecifier::resolve(
+        "data:text/plain,export default 'kale';",
+        "http://deno.land/core/tests/006_url_imports.ts",
+      ).unwrap()
+      .to_string(),
+      "data:text/plain,export default 'kale';",
+    );
```

Comment:
This is a lot of repeated code. It would be good if you could use an array of tuples and iterate over them to do the asserts.

Reply:
Yep, will do :+1: